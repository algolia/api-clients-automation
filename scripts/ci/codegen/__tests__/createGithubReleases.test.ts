import { afterAll, describe, expect, it, vi } from 'vitest';
import { createGitHubRelease } from '../createGitHubReleases.ts';

describe('createGithubRelease', () => {
  afterAll(() => {
    vi.clearAllMocks();
  });

  it('generates a release', async () => {
    vi.stubEnv('GITHUB_TOKEN', 'token');

    const createRelease = vi.fn().mockResolvedValue(undefined);
    const octokit = {
      repos: {
        createRelease,
      },
    } as any;

    vi.mock('../../utils.ts', async (importOriginal) => {
      return {
        // eslint-disable-next-line
        ...(await importOriginal<typeof import('../../utils.ts')>()),
        cloneRepository: vi.fn().mockResolvedValue({ tempGitDir: __dirname }),
      };
    });

    vi.mock('../../../common.ts', async (importOriginal) => {
      return {
        // eslint-disable-next-line
        ...(await importOriginal<typeof import('../../../common.ts')>()),
        run: vi.fn().mockResolvedValue('5.20.0\n5.19.0\n5.18.0\n5.17.1'),
      };
    });

    await expect(createGitHubRelease(octokit, 'dart')).resolves.toBeUndefined();

    expect(createRelease).toMatchSnapshot();
  });
});
