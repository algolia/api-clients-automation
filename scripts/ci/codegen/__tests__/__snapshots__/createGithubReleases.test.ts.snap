// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createGithubRelease > generates a release 1`] = `
[MockFunction spy] {
  "calls": [
    [
      {
        "body": "
# New version released!
## What's Changed
* [22167b8e68](https://github.com/algolia/api-clients-automation/commit/22167b8e68) fix(specs): extend Analytics descriptions ([#4360](https://github.com/algolia/api-clients-automation/pull/4360)) by @gazconroy
* [5b14906c1e](https://github.com/algolia/api-clients-automation/commit/5b14906c1e) fix(specs): remove the processing outcome ([#4359](https://github.com/algolia/api-clients-automation/pull/4359)) by @millotp
* [952639f0fb](https://github.com/algolia/api-clients-automation/commit/952639f0fb) fix(specs): adjust style for banner descriptions ([#4362](https://github.com/algolia/api-clients-automation/pull/4362)) by @kai687
* [4ae9ccd89f](https://github.com/algolia/api-clients-automation/commit/4ae9ccd89f) chore(deps): dependencies 2025-01-20 ([#4365](https://github.com/algolia/api-clients-automation/pull/4365)) by @algolia-bot
* [83f188d333](https://github.com/algolia/api-clients-automation/commit/83f188d333) feat(specs): rename composition to composition-full (private) and add composition (public) ([#4357](https://github.com/algolia/api-clients-automation/pull/4357)) by @e-krebs
* [e842e63d4e](https://github.com/algolia/api-clients-automation/commit/e842e63d4e) fix(javascript): use beta version ([#4385](https://github.com/algolia/api-clients-automation/pull/4385)) by @shortcuts

**Full Changelog**: https://github.com/algolia/algoliasearch-client-dart/compare/5.19.0...5.20.0

â†’ [Browse the Algolia documentation](https://www.algolia.com/doc/libraries/dart)
",
        "name": "5.20.0",
        "owner": "algolia",
        "repo": "algoliasearch-client-dart",
        "tag_name": "5.20.0",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
